### PR TITLE
feat(*): add close upto

### PR DESCRIPTION
**Pull request by chetan-fueled**
- We should be able to close upto a certain Fragment identifier.
- Add `closeUpto(id)` and `closeUptoWithResults(resultResponse, id)` similar to `close(id)` and `closeWithResults(resultResponse, id)`. Here `id` is the identifier returned when `fragmentTransaction.commit()` is called.

### CODE REVIEW

**Comment by chetan-fueled** [comment]
@husseinala I'm calling `displayFragmentForResults` in `HomeFragment` and `FirstFragment` and have used `backStackIdentifier` and `sourceFragmentId` static fields. Please tell if there is a better way of doing this.

File: flowr/src/main/java/com/fueled/flowr/Flowr.java [hunk 1]
```diff
@@ -282,15 +287,19 @@ protected final String getTagPrefix() {
...
                 setCurrentFragment(fragment);
             }
 
+            if (data.getTargetFragmentId() != null) {
+                targetFragmentId = data.getTargetFragmentId();
+            }
         } catch (Exception e) {
             Log.e(TAG, "Error while displaying fragment.", e);
         }
+        return identifier;
```
Comment (ragdroid) [inline flowr/src/main/java/com/fueled/flowr/Flowr.java]: Kindly add documentation on the newly returned value from the function.

File: flowr/src/main/java/com/fueled/flowr/Flowr.java [hunk 2]
```diff
@@ -402,6 +411,26 @@ public void close(int n) {
         }
     }
 
+    /**
+     * Closes the current activity if the fragments back stack is empty,
```
Comment (ragdroid) [inline flowr/src/main/java/com/fueled/flowr/Flowr.java]: documentation added says that it pops top `n` fragments, whereas we are popping the stack upto the transaction identified `id`

File: flowr/src/main/java/com/fueled/flowr/Flowr.java [hunk 3]
```diff
@@ -740,9 +784,11 @@ public Builder noTransactionAnimation() {
 
         /**
          * Displays the fragment using this builder configurations.
+         *
+         * @return id Identifier of the stated to be popped.
```
Comment (ragdroid) [inline flowr/src/main/java/com/fueled/flowr/Flowr.java]: Kindly modify the documentation to Identifier returned by `transaction.commit()`, to be used along with `#closeUptoWithResults()`

File: flowr/src/main/java/com/fueled/flowr/Flowr.java [hunk 4]
```diff
@@ -752,8 +798,10 @@ public void displayFragment() {
          *                    it will be later used to deliver the results to the correct fragment instance.
          * @param requestCode this code will be returned in {@link ResultResponse} when the fragment is closed,
          *                    and it can be used to identify the request from which the results were returned.
+         *
+         * @return id Identifier of the stated to be popped.
```
Comment (ragdroid) [inline flowr/src/main/java/com/fueled/flowr/Flowr.java]: to modify doc same as last one

File: flowr/src/main/java/com/fueled/flowr/internal/TransactionData.java [hunk 5]
```diff
@@ -121,4 +122,12 @@ public int getExitAnim() {
     public void setExitAnim(int exitAnim) {
         this.exitAnim = exitAnim;
     }
+
+    public String getTargetFragmentId() {
```
Comment (ragdroid) [inline flowr/src/main/java/com/fueled/flowr/internal/TransactionData.java]: 👍

File: sample/src/main/java/com/fueled/flowr/sample/FirstFragment.java [hunk 6]
```diff
@@ -2,32 +2,42 @@
...
 
     @Override
     public NavigationIconType getNavigationIconType() {
         return NavigationIconType.HAMBURGER;
     }
+
+    @Override
+    public void onClick(View view) {
+        getFlowr().open("/second")
+                .displayFragmentForResults(Flowr.getTargetFragmentId(), HomeFragment.RC_STACK);
```
Comment (ragdroid) [inline sample/src/main/java/com/fueled/flowr/sample/FirstFragment.java]: `targetFragmentId`
Do we need this to be a `static` variable. we can use this as an instance variable instead 🤔

File: sample/src/main/java/com/fueled/flowr/sample/HomeFragment.java [hunk 7]
```diff
@@ -57,4 +67,19 @@ private void displayLinkFragment() {
         getFlowr().open("/hello")
                 .displayFragment();
     }
+
+    private void displayFirstFragment() {
+        backStackIdentifier = getFlowr().open("/first")
+                .targetFragmentId(getFragmentId())
```
Comment (ragdroid) [inline sample/src/main/java/com/fueled/flowr/sample/HomeFragment.java]: and I think it's the `sourceFragmentId` not `target` 🤔

Comment (chetan-fueled) [reply sample/src/main/java/com/fueled/flowr/sample/HomeFragment.java]: `targetFragmentId` is inspired from `fragment.setTargetFragment(fragment, requestCode)` which specifies the target fragment for results. As per docs `@param fragment The fragment that is the target of this one.`

File: sample/src/main/java/com/fueled/flowr/sample/FirstFragment.java (hunk 6 above)
Comment (chetan-fueled) [reply sample/src/main/java/com/fueled/flowr/sample/FirstFragment.java]: Yes we can do `getFlowr().getTargetFragmentId()` but that didn't look very nice. I'm confused about this.

File: flowr/src/main/java/com/fueled/flowr/Flowr.java (hunk 4 above)
Comment (chetan-fueled) [reply flowr/src/main/java/com/fueled/flowr/Flowr.java]: Done! Thanks 👍

File: flowr/src/main/java/com/fueled/flowr/Flowr.java (hunk 3 above)
Comment (chetan-fueled) [reply flowr/src/main/java/com/fueled/flowr/Flowr.java]: Done! Thanks 👍

File: flowr/src/main/java/com/fueled/flowr/Flowr.java (hunk 2 above)
Comment (chetan-fueled) [reply flowr/src/main/java/com/fueled/flowr/Flowr.java]: Done! Thanks 👍

File: flowr/src/main/java/com/fueled/flowr/Flowr.java (hunk 1 above)
Comment (chetan-fueled) [reply flowr/src/main/java/com/fueled/flowr/Flowr.java]: Done! Thanks 👍

**Pull request merged**